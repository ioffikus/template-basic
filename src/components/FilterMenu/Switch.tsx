import React from 'react';
import { Switch as AntdSwitch } from 'antd';

interface IProps {
  value: boolean;
  fieldSlug: string;
  valueType: string;
  filterType: string;
  onChange: (value: boolean, fieldSlug: string, valueType: string, filterType: string) => void;
}

interface IState {}

class Switch extends React.PureComponent<IProps, IState> {
  handleChange = () => {
    this.props.onChange(!this.props.value, this.props.fieldSlug, this.props.valueType, this.props.filterType);
  };

  render() {
    return (
      <div className="root">
        <AntdSwitch checked={!!this.props.value} onChange={this.handleChange} />
      </div>
    );
  }
}

export default Switch;
