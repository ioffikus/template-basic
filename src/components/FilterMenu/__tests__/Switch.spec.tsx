import React from 'react';
import { shallow } from 'enzyme';
import { Switch as AntdSwitch } from 'antd';
import filterStateMock from 'src/store/__mocks__/filterStore';
import Switch from '../Switch';

describe(null, () => {
  it('render() correct with initialState', () => {
    const switchMock = filterStateMock.results[1];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = switchMock;
    const value: boolean = null;
    const onChange = jest.fn();
    const wrapper = shallow(
      <Switch value={value} fieldSlug={fieldSlug} valueType={valueType} filterType={filterType} onChange={onChange} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('render() correct with value=true', () => {
    const switchMock = filterStateMock.results[1];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = switchMock;
    const value = true;
    const onChange = jest.fn();
    const wrapper = shallow(
      <Switch value={value} fieldSlug={fieldSlug} valueType={valueType} filterType={filterType} onChange={onChange} />,
    );
    const switchInstance = wrapper
      .find(AntdSwitch)
      .dive()
      .instance() as AntdSwitch;
    expect(switchInstance.props.checked).toEqual(value);
    expect(wrapper).toMatchSnapshot();
  });
  it('AntdSwitch.onChange()', () => {
    const switchMock = filterStateMock.results[1];
    const { field_slug: fieldSlug, value_type: valueType, filter_type: filterType } = switchMock;
    const value: boolean = null;
    const newValue = true;
    const onChange = jest.fn();
    const wrapper = shallow(
      <Switch value={value} fieldSlug={fieldSlug} valueType={valueType} filterType={filterType} onChange={onChange} />,
    );
    const switchInstance = wrapper
      .find(AntdSwitch)
      .dive()
      .instance() as AntdSwitch;
    switchInstance.props.onChange({ target: { value: newValue } } as any);
    expect(onChange).toBeCalledWith(newValue, fieldSlug, valueType, filterType);
  });
});
