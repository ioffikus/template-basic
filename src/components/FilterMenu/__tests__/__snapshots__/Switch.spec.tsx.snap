// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correct with initialState 1`] = `
<div
  className="root"
>
  <Switch
    checked={false}
    onChange={[Function]}
    prefixCls="ant-switch"
  />
</div>
`;

exports[`render() correct with value=true 1`] = `
<div
  className="root"
>
  <Switch
    checked={true}
    onChange={[Function]}
    prefixCls="ant-switch"
  />
</div>
`;
